app/controllers/application_controller.rb

```diff
@@ -269,4 +269,19 @@ def domain_redirection_required?
   def redirect_to_primary_domain
     observable_redirect_to "#{request.ssl? ? 'https' : 'http'}://#{current_school.domains.primary.fqdn}#{request.path}"
   end
+
+  before_action :set_last_seen_at,
+                if:
+                  proc {
+                    user_signed_in? &&
+                      (
+                        session[:last_seen_at] == nil ||
+                          session[:last_seen_at] < 15.minutes.ago
```

Comment:
This will probably not work as expected because these session values are stored in cookies, which only support strings.

Here's a quick demo:

```ruby
session[:foo] ||= Time.zone.now

# raise "Foo is a #{session[:foo].class}" if session.key?(:foo)
```

Run the first line in an action, and then only the second line on the next request, to read the stored session value.

It'll say that `Foo is a String` because cookies can only store strings, and Rails will automatically serialize the supplied value before storing it in the cookie.

If you run both lines together, note that it'll say that `Foo is a ActiveSupport::TimeWithZone` - that won't apply to regular requests because the setting and getting of the value won't happen in the same request.

This needs to be manually serialized and deserialized for safety. Example:

```ruby
# Convert to ISO 8601.
session[:foo] ||= Time.zone.now.iso8601

# Parse the string value when reading stored timestamp.
if session.key?(:foo)
  raise "Foo is a #{session[:foo].class} and its parsed value is #{Time.zone.parse(session[:foo])}"
end
```